__tests__/integration/nanocontracts/bet.test.js

```diff
@@ -0,0 +1,380 @@
+import {
+  generateWalletHelper,
+  stopAllWallets,
+  waitForTxReceived,
+  waitUntilNextTimestamp,
+  waitNextBlock,
+  waitTxConfirmed
+} from '../helpers/wallet.helper';
+import { HATHOR_TOKEN_CONFIG } from '../../../src/constants';
+import ncApi from '../../../src/api/nano';
+import helpersUtils from '../../../src/utils/helpers';
+import dateFormatter from '../../../src/utils/date';
+import { bufferToHex, hexToBuffer } from '../../../src/utils/buffer';
+import Address from '../../../src/models/address';
+import P2PKH from '../../../src/models/p2pkh';
+import { isEmpty } from 'lodash';
+import { getOracleBuffer, getOracleInputData } from '../../../src/nano_contracts/utils';
+import Serializer from '../../../src/nano_contracts/serializer';
+import { NanoContractTransactionError, NanoRequest404Error } from '../../../src/errors';
+
+let fundsTx;
+
+describe('full cycle of bet nano contract', () => {
+  /** @type HathorWallet */
+  let hWallet;
+
+  beforeAll(async () => {
+    hWallet = await generateWalletHelper();
+    fundsTx = await GenesisWalletHelper.injectFunds(hWallet, await hWallet.getAddressAtIndex(0), 1000);
+    await waitForTxReceived(hWallet, fundsTx.hash);
+  });
+
+  afterAll(async () => {
+    await hWallet.stop();
+    await GenesisWalletHelper.clearListeners();
+  });
+
+  const checkTxValid = async (txId) => {
+    expect(txId).toBeDefined();
+    await waitForTxReceived(hWallet, txId);
+    // We need to wait for the tx to get a first block, so we guarantee it was executed
+    await waitTxConfirmed(hWallet, txId);
+    // Now we query the transaction from the full node to double check it's still valid after the nano execution
+    // and it already has a first block, so it was really executed
+    const txAfterExecution = await hWallet.getFullTxById(txId);
+    expect(isEmpty(txAfterExecution.meta.voided_by)).toBe(true);
+    expect(isEmpty(txAfterExecution.meta.first_block)).not.toBeNull();
+  }
+
+  it('bet deposit', async () => {
+    const address0 = await hWallet.getAddressAtIndex(0);
+    const address1 = await hWallet.getAddressAtIndex(1);
+    const dateLastBet = dateFormatter.dateToTimestamp(new Date()) + 6000;
+    const network = hWallet.getNetworkObject();
+    const blueprintId = '3cb032600bdf7db784800e4ea911b10676fa2f67591f82bb62628c234e771595';
+
+    // Create NC
+    const oracleData = getOracleBuffer(address1, network);
+    const tx1 = await hWallet.createAndSendNanoContractTransaction(
+      'initialize',
+      address0,
+      {
+        blueprintId,
+        args: [
+          bufferToHex(oracleData),
+          HATHOR_TOKEN_CONFIG.uid,
+          dateLastBet
+        ]
+      }
+    );
+    await checkTxValid(tx1.hash);
+
+    // Bet 100 to address 2
+    const address2 = await hWallet.getAddressAtIndex(2);
+    const address2Obj = new Address(address2, { network });
+    const txBet = await hWallet.createAndSendNanoContractTransaction(
+      'bet',
```

Comment:
We have constants for these, don't we?

Reply:
No, the method names are different for each blueprint (only initialize is the same)